qcengine/procedures/berny.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+import traceback
+from typing import Any, Dict, Union
+import pkg_resources
+
+import qcengine
+from qcelemental.models import OptimizationInput, OptimizationResult
+from qcelemental.util import which_import
+
+from .model import ProcedureHarness
+from ..config import TaskConfig
+
+# TODO there is a lot of boilerplate shared with GeometricProcedure
+# TODO how to handle logging?
+# TODO where to store paramters for Berny?
+
+
+class BernyProcedure(ProcedureHarness):
+
+    _defaults = {"name": "Berny", "procedure": "optimization"}
+
+    class Config(ProcedureHarness.Config):
+        pass
+
+    def found(self, raise_error: bool = False) -> bool:
+        return which_import(
+            "berny", return_bool=True, raise_error=raise_error, raise_msg="Please install via `pip install pyberny`.",
+        )
+
+    def build_input_model(self, data: Union[Dict[str, Any], "OptimizationInput"]) -> "OptimizationInput":
+        return self._build_model(data, OptimizationInput)
+
+    def compute(self, input_data: "OptimizationInput", config: "TaskConfig") -> "OptimizationResult":
+        try:
+            import berny
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError("Could not find Berny in the Python path.")
+        input_data = input_data.dict()
+        geom_qcng = input_data["initial_molecule"]
+        comput = {**input_data["input_specification"], "molecule": geom_qcng}
+        trajectory = []
+        output_data = input_data.copy()
+        try:
+            geom_berny = berny.Geometry(geom_qcng["symbols"], geom_qcng["geometry"])
+            opt = berny.Berny(geom_berny)
+            for geom_berny in opt:
+                geom_qcng["geometry"] = np.stack(geom_berny.coords)
+                ret = qcengine.compute(comput, **input_data["keywords"])
```

Comment:
`from ..compute import compute`. I think you will also need a `input_data["keywords"]["program"]` here.

Reply:
`input_data["keywords"]` already contains `{"program": ...}`, so that's why this works. I thought that `input_data["keywords"]` is supposed to be passed to `qcengine.compute()` in this way. Is it?

Reply:
So is [this](https://github.com/MolSSI/QCEngine/pull/232/commits/7ef994ede43cac838e119d59eb1e46ce2fec53ba#diff-4384465c212e54921f44238616ea614aR44) how it should be done? Passing only `"program"` explicitly.